crates/sc-subspace-block-relay/src/lib.rs

```diff
@@ -153,3 +159,32 @@ struct ProtocolUnitInfo<ProtocolUnitId, ProtocolUnit> {
     /// action is needed on client side to resolve it
     unit: Option<ProtocolUnit>,
 }
+
+/// The relay protocol identifier.
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode)]
+pub(crate) enum RelayProtocol {
+    #[codec(index = 0)]
+    CompactBlock,
+    // Next protocol goes here.
+    // #[codec(index = 1)]
+}
+
+/// The relay protocol version.
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode)]
+pub(crate) struct RelayVersion {
+    /// The protocol identifier.
+    protocol: RelayProtocol,
+
+    /// The sub version within the protocol implementation.
+    protocol_version: u64,
```

Comment:
What purpose does it serve? We can always introduce CompactBlockV2 or add things to other data structures, I do not get the use case of this.

It is also not checked anywhere anyway, feels like unnecessary complexity.

Reply:
This commit introduces the types used in later commit, updated the commit message to reflect it